MariaDB/server
# MDEV-32935: Remove unneeded CMAKE_SYSTEM_PROCESSOR parameter from debian/rules

- [x] *The Jira issue number for this PR is: MDEV-32935*

## Description
There is no need for CMAKE_SYSTEM_PROCESSOR parameter in Debian build as dh_auto_configure should handle things better and more reliable

## How can this PR be tested?
This is bit tricky to test. One can make crosscompile toolchain on Debian Sid manually with:
```
dpkg --add-architecture arm64

apt install git-buildpackage bison cmake debhelper dh-exec libboost-dev libbz2-dev libcrack2-dev libcurl4-openssl-dev libedit-dev libedit-dev:amd64 libfmt-dev libjemalloc-dev libjudy-dev libkrb5-dev liblz4-dev liblzma-dev liblzo2-dev libncurses-dev libnuma-dev libpam0g-dev libpcre2-dev libpmem-dev libsnappy-dev libssl-dev libssl-dev:amd64 libsystemd-dev liburing-dev libxml2-dev libzstd-dev lsb-release perl:any po-debconf psmisc unixodbc-dev uuid-dev zlib1g-dev zlib1g-dev:amd64 libc-dev libstdc++-13-dev build-essential:amd64 fakeroot:amd64 crossbuild-essential-arm64:amd64

apt install default-jdk:arm64 libaio-dev:arm64 libboost-dev:arm64 libbz2-dev:arm64 libcrack2-dev:arm64 libedit-dev:arm64 libfmt-dev:arm64 libjemalloc-dev:arm64 libjudy-dev:arm64 libkrb5-dev:arm64 liblz4-dev:arm64 liblzma-dev:arm64 liblzo2-dev:arm64 libnuma-dev:arm64 libpam0g-dev:arm64 libpcre2-dev:arm64 libpmem-dev:arm64 libsnappy-dev:arm64 libssl-dev:arm64 libsystemd-dev:arm64 liburing-dev:arm64 libxml2-dev:arm64 libzstd-dev:arm64 unixodbc-dev:arm64 uuid-dev:arm64 zlib1g-dev:arm64
```

And then build it with:
```
dpkg-buildpackage --sanitize-env -aarm64 -Pcross,nocheck -us -uc -B -rfakeroot --jobs-try=1
```

It should work as expected.

## Basing the PR against the correct MariaDB version

## PR quality check
